lib/console/deployments/pr/impl/gitlab.ex

```diff
@@ -70,7 +70,67 @@ defmodule Console.Deployments.Pr.Impl.Gitlab do
     end
   end
 
-  def files(_, _), do: {:ok, []}
+  def files(conn, url) do
+    with {:ok, group, number} <- get_pull_id(url),
+         {:ok, conn} <- connection(conn),
+         {:ok, %HTTPoison.Response{status_code: 200, body: body}} <- HTTPoison.get("#{conn.host}/api/v4/projects/#{uri_encode(group)}/merge_requests/#{number}/changes", Connection.headers(conn)),
+         {:ok, %{"changes" => changes}} <- Jason.decode(body) do
+
+      files = Enum.map(changes, fn change ->
+        # Get full file content for the new_path if it's not deleted
+        file_content = if change["deleted_file"] do
+          ""
+        else
+          # Get the full file content from the repository files API
+          case get_file_content(conn, group, change["new_path"], "HEAD") do
```

Comment:
I'd just have these functions take the fallback as the final arg and then return them out of the internal case statement so you don't have to wrap it in this func